rct229/rulesets/ashrae9012019/section5/section5rule38.py

```diff
@@ -0,0 +1,251 @@
+                    index_rmr="baseline",
+                    list_path="subsurfaces[*]",
+                    manual_check_required_msg=MANUAL_CHECK_MSG,
+                )
+
+            def create_data(self, context, data=None):
+                surface_b = context.baseline
+                surface_id_b = surface_b["id"]
+
+                return {"surface_id_b": surface_id_b}
+
+            class SubsurfaceRule(RuleDefinitionBase):
+                def __init__(self):
+                    super(
+                        Section5Rule38.BuildingRule.RoofRule.SubsurfaceRule, self
+                    ).__init__(
+                        rmrs_used=UserBaselineProposedVals(False, True, False),
+                        required_fields={
+                            "$": ["classification", "glazed_area", "opaque_area"]
+                        },
+                    )
+
+                def is_applicable(self, context, data=None):
+                    subsurface_b = context.baseline
+
+                    return (
+                        subsurface_b.get("classification")
+                        == SURFACE_CLASSIFICATION.DOOR
+                        and subsurface_b.get("glazed_area")
+                        > subsurface_b.get("opaque_area")
+                    ) or subsurface_b.get(
+                        "classification"
+                    ) != SURFACE_CLASSIFICATION.DOOR
+
+                def get_calc_vals(self, context, data=None):
+                    subsurface_b = context.baseline
+                    subsurface_b_shgc = subsurface_b["solar_heat_gain_coefficient"]
+
+                    subsuface_type_b = data["surface_conditioning_category_dict"][
+                        data["surface_id_b"]
+                    ]
+                    target_shgc_res_b = data["target_shgc_res"]
+                    target_shgc_nonres_b = data["target_shgc_nonres"]
+                    target_shgc_semiheated_b = data["target_shgc_semiheated"]
+                    target_shgc = 0.0
+
+                    if (
+                        subsuface_type_b == SCC.EXTERIOR_MIXED
+                        or subsuface_type_b == SCC.EXTERIOR_RESIDENTIAL
+                    ):
+                        target_shgc = target_shgc_res_b
+                    elif subsuface_type_b == SCC.EXTERIOR_NON_RESIDENTIAL:
+                        target_shgc = target_shgc_nonres_b
+                    else:
+                        target_shgc = target_shgc_semiheated_b
+
+                    return {
+                        "subsurface_b_shgc": subsurface_b_shgc,
+                        "target_shgc": target_shgc,
+                    }
+
+                def rule_check(self, context, calc_vals=None, data=None):
+                    subsurface_b_shgc = calc_vals["subsurface_b_shgc"]
+                    target_shgc = calc_vals["target_shgc"]
+
+                    return std_equal(subsurface_b_shgc, target_shgc)
```

Comment:
The first argument of the `std_equal` is `std_val`
Though the order does not change the outcome, I suggest keeping it consistent:
`std_equal(target_shgc, subsurface_shgc_b`

Reply:
Fixed.